gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/sampling/PageRankSampling.java

```diff
@@ -0,0 +1,161 @@
+
+/**
+ * Computes a PageRank-Sampling of the graph.
+ *
+ * Uses the Gradoop-Wrapper of Flinks PageRank-algorithm {@link PageRank} with a dampening factor
+ * and a number of maximum iterations. It computes a per-vertex score which is the sum of the
+ * PageRank-scores transmitted over all in-edges. The score of each vertex is divided evenly
+ * among its out-edges.
+ *
+ * If vertices got different PageRank-scores, all scores are scaled in a range between 0 and 1.
+ * Then it retains all vertices with a PageRank-score greater or equal/smaller than a given
+ * sampling threshold - depending on the Boolean set in {@code sampleGreaterThanThreshold}.
+ *
+ * If ALL vertices got the same PageRank-score, it can be decided whether to sample all vertices
+ * or none of them - depending on the Boolean set in {@code keepVerticesIfSameScore}.
+ *
+ * Retains all edges which source- and target-vertices were chosen. There may retain some
+ * unconnected vertices in the sampled graph.
+ */
+public class PageRankSampling extends SamplingAlgorithm {
+
+  /**
+   * Dampening factor used by PageRank-algorithm
+   */
+  private final double dampeningFactor;
+  /**
+   * Number of iterations used by PageRank-algorithm
+   */
+  private final int maxIteration;
+  /**
+   * Sampling threshold for PageRankScore
+   */
+  private final double threshold;
+  /**
+   * Whether to sample vertices with PageRank-score greater (true) or equal/smaller (false)
+   * than the threshold
+   */
+  private final boolean sampleGreaterThanThreshold;
+  /**
+   * Whether to sample all vertices (true) or none of them (false), in case all vertices got the
+   * same PageRank-score.
+   */
+  private final boolean keepVerticesIfSameScore;
+
+  /**
+   * Creates a new PageRankSampling instance.
+   *
+   * @param dampeningFactor The dampening factor used by PageRank-algorithm, e.g. 0.85
+   * @param maxIteration The number of iterations used by PageRank-algorithm, e.g. 40
+   * @param threshold The threshold for the PageRank-score (ranging between 0 and 1 when scaled),
+   *                  determining if a vertex is sampled, e.g. 0.5
+   * @param sampleGreaterThanThreshold Whether to sample vertices with a PageRank-score
+   *                                   greater (true) or equal/smaller (false) the threshold
+   * @param keepVerticesIfSameScore Whether to sample all vertices (true) or none of them (false)
+   *                                in case all vertices got the same PageRank-score.
+   */
+  public PageRankSampling(double dampeningFactor, int maxIteration, double threshold,
+    boolean sampleGreaterThanThreshold, boolean keepVerticesIfSameScore) {
+    this.dampeningFactor = dampeningFactor;
+    this.threshold = threshold;
+    this.maxIteration = maxIteration;
+    this.sampleGreaterThanThreshold = sampleGreaterThanThreshold;
+    this.keepVerticesIfSameScore = keepVerticesIfSameScore;
+  }
+
+  /**
+   * {@inheritDoc}
```

Comment:
inheriting results in a comment `Creates a new graph from a randomly chosen subset of nodes and their associated edges.` ... That fits not directly. Either write a own Description or extend the inherit with the note that the PR-Sampling is used here.